server/mailrepository/mailrepository-blob/src/test/java/org/apache/james/mailrepository/blob/BlobMailRepositoryTest.java

```diff
@@ -56,4 +58,15 @@ void setup() {
     public MailRepository retrieveRepository() {
         return blobMailRepository;
     }
+
+    @Override
+    public MailRepository retrieveRepository(MailRepositoryPath path) {
+        return blobMailRepository;
```

Comment:
I don't actually agree with the test setup. We intend to instantiate a mailrepository per mailrepository url and use the url to define the bucket to use. as such the retrieveRepository(path) would return a different repository backed by a different bucket for a different path

Reply:
Note that this is exactly what you do in the casandra implementatio of the test, you return a different repository for a different path ..

Reply:
Current code injecting mailRepository passes the URL to implementations via either config or constructor, and shoud have the underlying mail repository adapt its storage to the URL.

I'll adapt the implem.

Regards